perfkitbenchmarker/linux_virtual_machine.py

```diff
@@ -198,8 +221,29 @@ def PrepareVMEnvironment(self):
       if self.is_static:
         self.SnapshotPackages()
       self.SetupPackageManager()
+    self.ConfigureVMKernel()
     self.BurnCpu()
 
+  def ConfigureVMKernel(self):
+    """Apply {procfs,sysfs}_config to VM."""
+
+    path_components = None
+
+    def ConfigureSubtree(subtree):
+      for name, value in subtree.iteritems():
+        path_components.append(name)
+        if isinstance(value, basestring):
```

Comment:
PyYaml will automagically cast string numbers into numbers for you:

```
$ cat /tmp/sample.yaml 
a: {b: c}
d: {e: 4}

In [6]: y=yaml.load(open('/tmp/sample.yaml'))

In [7]: y
Out[7]: {'a': {'b': 'c'}, 'd': {'e': 4}}
```

Note that the value for "e" is the number 4.  maybe check that the value is *not* another dict?